trino/sqlalchemy/compiler.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from sqlalchemy.sql import compiler
+
+# https://trino.io/docs/current/language/reserved.html
+RESERVED_WORDS = {
+    "alter",
+    "and",
+    "as",
+    "between",
+    "by",
+    "case",
+    "cast",
+    "constraint",
+    "create",
+    "cross",
+    "cube",
+    "current_date",
+    "current_path",
+    "current_role",
+    "current_time",
+    "current_timestamp",
+    "current_user",
+    "deallocate",
+    "delete",
+    "describe",
+    "distinct",
+    "drop",
+    "else",
+    "end",
+    "escape",
+    "except",
+    "execute",
+    "exists",
+    "extract",
+    "false",
+    "for",
+    "from",
+    "full",
+    "group",
+    "grouping",
+    "having",
+    "in",
+    "inner",
+    "insert",
+    "intersect",
+    "into",
+    "is",
+    "join",
+    "left",
+    "like",
+    "localtime",
+    "localtimestamp",
+    "natural",
+    "normalize",
+    "not",
+    "null",
+    "on",
+    "or",
+    "order",
+    "outer",
+    "prepare",
+    "recursive",
+    "right",
+    "rollup",
+    "select",
+    "table",
+    "then",
+    "true",
+    "uescape",
+    "union",
+    "unnest",
+    "using",
+    "values",
+    "when",
+    "where",
+    "with",
+}
+
+
+class TrinoSQLCompiler(compiler.SQLCompiler):
+    pass
+
+
+class TrinoDDLCompiler(compiler.DDLCompiler):
+    pass
+
+
+class TrinoTypeCompiler(compiler.GenericTypeCompiler):
+    def visit_FLOAT(self, type_, **kw):
+        precision = type_.precision or 32
+        if 0 <= precision <= 32:
+            return self.visit_REAL(type_, **kw)
+        elif 32 < precision <= 64:
+            return self.visit_DOUBLE(type_, **kw)
+        else:
+            raise ValueError(f"type.precision must be in range [0, 64], got {type_.precision}")
```

Comment:
It's because `GenericTypeCompiler` has default impl for `visit_FLOAT`.  
Other behavior is `raise Exception("not support FLOAT data type")`, so which one you prefer?

Reply:
Discussed offline. The reason it is this way is because SqlAlchemy has FLOATS that can be 64 bits and we are only mapping to REAL or DOUBLE depending on what precision is provided.